Other/MergeShops/EctocaveMerge.cs

```diff
@@ -0,0 +1,154 @@
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface Bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Ichorus Scythe Piece", "Ichorus Egg", "Pure Ichor Gem", "Slime", "Dragon Rogue", "Dragon Rogue Klinge", "Dragon Rogue Hood", "Dragon Rogue Twin Klinge Cape", "Piece of Fabric", "Bone"});
+        Core.SetOptions();
+
+        BuyAllMerge();
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge(string buyOnlyThis = null, mergeOptionsEnum? buyMode = null)
+    {
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("ectocave", 1039, findIngredients, buyOnlyThis, buyMode: buyMode);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Ichorus Scythe Piece":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.EquipClass(ClassType.Farm);
+                    Core.RegisterQuests(3874);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("ectocave", "Ichor Dracolich", "Uncut Emerald", 9, log: false);
+                        Core.HuntMonster("ectocave", "Ektorax", "Brilliant Diamond", log: false);
+                        Bot.Wait.ForPickup(req.Name);
+                    }
+                    Core.CancelRegisteredQuests();
+                    break;
+
+                case "Ichorus Egg":
+                    if (!Bot.Player.IsMember)
+                        break;
+
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.EquipClass(ClassType.Farm);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("ectocave", "Ektorax", "Ichorus Egg");
```

Comment:
`Core.HuntMonster("ectocave", "Ektorax", "Ichorus Egg");`

should be

`Core.HuntMonster("ectocave", "Ektorax");`